### ISSUE TITLE
loss = nan

### ISSUE THREAD

**Issue by luuuyi** [D-X-Y/caffe-faster-rcnn#39]
你好，我在使用你的脚本train的时候遇到如下问题，不知道作者您有遇到过么
I0115 14:49:27.785917 12606 solver.cpp:224] Iteration 0 (-1.27518e-43 iter/s, 0.804229s/20 iters), loss = 5.06628  [ 0 / 70000 ] -> [ -1:60 (H:M) ]
I0115 14:49:27.786006 12606 solver.cpp:243]     Train net output #0: bbox_accuracy = 0.046875
I0115 14:49:27.786028 12606 solver.cpp:243]     Train net output #1: bbox_loss = 0.369181 (* 1 = 0.369181 loss)
I0115 14:49:27.786036 12606 solver.cpp:243]     Train net output #2: cls_loss = 3.12335 (* 1 = 3.12335 loss)
I0115 14:49:27.786043 12606 solver.cpp:243]     Train net output #3: rpn_cls_loss = 0.895593 (* 1 = 0.895593 loss)
I0115 14:49:27.786051 12606 solver.cpp:243]     Train net output #4: rpn_loss_bbox = 1.12812 (* 1 = 1.12812 loss)
I0115 14:49:27.786078 12606 sgd_solver.cpp:105] Iteration 0, lr = 0.001
I0115 14:49:33.189517 12606 solver.cpp:224] Iteration 20 (3.70129 iter/s, 5.40352s/20 iters), loss = nan  [ 20 / 70000 ] -> [ 3:36 (H:M) ]
I0115 14:49:33.189579 12606 solver.cpp:243]     Train net output #0: bbox_accuracy = 1
I0115 14:49:33.189594 12606 solver.cpp:243]     Train net output #1: bbox_loss = nan (* 1 = nan loss)
I0115 14:49:33.189606 12606 solver.cpp:243]     Train net output #2: cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:33.189617 12606 solver.cpp:243]     Train net output #3: rpn_cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:33.189627 12606 solver.cpp:243]     Train net output #4: rpn_loss_bbox = nan (* 1 = nan loss)
I0115 14:49:33.189640 12606 sgd_solver.cpp:105] Iteration 20, lr = 0.001
I0115 14:49:38.072799 12606 solver.cpp:224] Iteration 40 (4.09564 iter/s, 4.88324s/20 iters), loss = nan  [ 40 / 70000 ] -> [ 3:59 (H:M) ]
I0115 14:49:38.072854 12606 solver.cpp:243]     Train net output #0: bbox_accuracy = 1
I0115 14:49:38.072875 12606 solver.cpp:243]     Train net output #1: bbox_loss = nan (* 1 = nan loss)
I0115 14:49:38.072883 12606 solver.cpp:243]     Train net output #2: cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:38.072891 12606 solver.cpp:243]     Train net output #3: rpn_cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:38.072898 12606 solver.cpp:243]     Train net output #4: rpn_loss_bbox = nan (* 1 = nan loss)
I0115 14:49:38.072906 12606 sgd_solver.cpp:105] Iteration 40, lr = 0.001
I0115 14:49:42.956974 12606 solver.cpp:224] Iteration 60 (4.09489 iter/s, 4.88414s/20 iters), loss = nan  [ 60 / 70000 ] -> [ 3:59 (H:M) ]
I0115 14:49:42.957036 12606 solver.cpp:243]     Train net output #0: bbox_accuracy = 1
I0115 14:49:42.957048 12606 solver.cpp:243]     Train net output #1: bbox_loss = nan (* 1 = nan loss)
I0115 14:49:42.957058 12606 solver.cpp:243]     Train net output #2: cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:42.957067 12606 solver.cpp:243]     Train net output #3: rpn_cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:42.957075 12606 solver.cpp:243]     Train net output #4: rpn_loss_bbox = nan (* 1 = nan loss)
I0115 14:49:42.957083 12606 sgd_solver.cpp:105] Iteration 60, lr = 0.001
I0115 14:49:47.778499 12606 solver.cpp:224] Iteration 80 (4.14811 iter/s, 4.82148s/20 iters), loss = nan  [ 80 / 70000 ] -> [ 4:2 (H:M) ]
I0115 14:49:47.778566 12606 solver.cpp:243]     Train net output #0: bbox_accuracy = 1
I0115 14:49:47.778589 12606 solver.cpp:243]     Train net output #1: bbox_loss = nan (* 1 = nan loss)
I0115 14:49:47.778600 12606 solver.cpp:243]     Train net output #2: cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:47.778607 12606 solver.cpp:243]     Train net output #3: rpn_cls_loss = 87.3365 (* 1 = 87.3365 loss)
I0115 14:49:47.778614 12606 solver.cpp:243]     Train net output #4: rpn_loss_bbox = nan (* 1 = nan loss)

**Comment by mai00dou** [comment]
我有相同的问题，请问您解决了么

**Comment by stormyoung** [comment]
I met the same question, how do you solve this problem?

**Comment by luuuyi** [comment]
@mai00dou 还没有 感觉中间存在bug

**Comment by luuuyi** [comment]
@stormyoung sorry, not yet

**Comment by stormyoung** [comment]
@luuuyi Thank you. I think if it's because the input data was not  preprocessed?  I use VOC2007 dataset and pre-trained zf caffemodel according to Readme.
If I do not use pre-trained zf caffemodel, loss = nan will not occur, but the net still can not converge,

**Comment by stormyoung** [comment]
If you have processes, could you tell me? Thank you

**Comment by hhawary** [comment]
@luuuyi in my case the bbox_loss and rpn_loss_bbox was nan values but it works fine when i remove the Accuracy layer from the train_val.prototxt [https://github.com/D-X-Y/caffe-faster-rcnn/blob/dev/models/FRCNN/zf/train_val.prototxt](url)
